core/src/main/java/io/onetable/client/OneTableClient.java

```diff
@@ -167,83 +159,42 @@ private <COMMIT> Map<TableFormat, TableFormatSync> getFormatsToSyncIncrementally
         lastSyncMetadataByFormat.entrySet().stream()
             .filter(entry -> syncClientByFormat.containsKey(entry.getKey()))
-            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            .collect(
+                Collectors.toMap(
+                    entry -> syncClientByFormat.get(entry.getKey()),
+                    entry -> entry.getValue().get()));
 
     InstantsForIncrementalSync instantsForIncrementalSync =
         getMostOutOfSyncCommitAndPendingCommits(filteredSyncMetadataByFormat);
     IncrementalTableChanges incrementalTableChanges =
         source.extractTableChanges(instantsForIncrementalSync);
-    if (!incrementalTableChanges.getTableChanges().isEmpty()) {
-      for (Map.Entry<TableFormat, TableFormatSync> entry : syncClientByFormat.entrySet()) {
-        TableFormat tableFormat = entry.getKey();
-        TableFormatSync tableFormatSync = entry.getValue();
-        List<Instant> pendingSyncsByFormat =
-            filteredSyncMetadataByFormat
-                .get(tableFormat)
-                .map(OneTableMetadata::getInstantsToConsiderForNextSync)
-                .orElse(Collections.emptyList());
-        Set<Instant> pendingInstantsSet = new HashSet<>(pendingSyncsByFormat);
-        // extract the changes that happened since this format was last synced
-        List<TableChange> filteredTableChanges =
-            incrementalTableChanges.getTableChanges().stream()
-                .filter(
-                    change ->
-                        change
-                                .getTableAsOfChange()
-                                .getLatestCommitTime()
-                                .isAfter(
-                                    filteredSyncMetadataByFormat
-                                        .get(tableFormat)
-                                        .map(OneTableMetadata::getLastInstantSynced)
-                                        .get())
-                            || pendingInstantsSet.contains(
-                                change.getTableAsOfChange().getLatestCommitTime()))
-                .collect(Collectors.toList());
-        if (filteredTableChanges.isEmpty()) {
-          // no updates to sync
-          continue;
-        }
-        IncrementalTableChanges tableFormatIncrementalChanges =
-            IncrementalTableChanges.builder()
-                .tableChanges(filteredTableChanges)
-                .pendingCommits(incrementalTableChanges.getPendingCommits())
-                .build();
-        List<SyncResult> syncResultList =
-            tableFormatSync.syncChanges(tableFormatIncrementalChanges);
-        syncedTable =
-            filteredTableChanges.get(filteredTableChanges.size() - 1).getTableAsOfChange();
-        SyncResult latestSyncResult = syncResultList.get(syncResultList.size() - 1);
-        syncResultsByFormat.put(tableFormat, latestSyncResult);
-      }
+    if (incrementalTableChanges.getTableChanges().hasNext()) {
```

Comment:
we can remove this check here as we are doing this inside `TableFormatSync.syncChanges`